tests/GlobalFilter.test.ts

```diff
@@ -0,0 +1,120 @@
+    afterEach(() => {
+        GlobalFilter.reset();
+    });
+
+    it('Should provide Global Filter with the default value with get()', () => {
+        // Arrange
+        const defaultValue = '';
+
+        // Assert
+        expect(GlobalFilter.get()).toEqual(defaultValue);
+    });
+
+    it('Should set new Global Filter', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.get()).toEqual(testValue);
+    });
+
+    it('Should reset the Global Filter', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        const defaultValue = '';
+        GlobalFilter.set(testValue);
+        GlobalFilter.reset();
+
+        // Assert
+        expect(GlobalFilter.get()).toEqual(defaultValue);
+    });
+
+    it('Should indicate empty Global Filter by default', () => {
+        // Assert
+        expect(GlobalFilter.isEmpty()).toEqual(true);
+    });
+
+    it('Should indicate non-empty Global Filter after setting one', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.isEmpty()).toEqual(false);
+    });
+
+    it('Should match a string with Global Filter', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        const testString = 'This is a string with newGlobalFilter inside';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.includedIn(testString)).toEqual(true);
+    });
+
+    it('Should not match a string without Global Filter', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        const testString = 'This is a string without Global Filter';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.includedIn(testString)).toEqual(false);
+    });
+
+    it('Should remove Global Filter from the beginning of a string', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        const testStringBefore = 'This is a string with GF at the end newGlobalFilter';
+        const testStringAfter = 'This is a string with GF at the end';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.removeAsSubstringFrom(testStringBefore)).toEqual(testStringAfter);
+    });
+
+    it('Should remove Global Filter from the end of a string', () => {
+        // Arrange
+        const testValue = 'newGlobalFilter';
+        const testStringBefore = 'newGlobalFilter This is a string with GF at the beginning';
+        const testStringAfter = 'This is a string with GF at the beginning';
+        GlobalFilter.set(testValue);
+
+        // Assert
+        expect(GlobalFilter.removeAsSubstringFrom(testStringBefore)).toEqual(testStringAfter);
+    });
+
+    // it('Should remove Global Filter from the middle of a string', () => {});
+    // Not supported
```

Comment:
I'm puzzled by this comment.

Here is a modified version of `ACME.md` in the Tasks-Demo vault.

```text
# ACME

## Steps to world domination

- [ ] #task I have the global filter at  the start of the line
- [ ] I have the global filter in #task the middle of the line
- [ ] I have the global filter at the end of the line #task

## Model test

- [ ] This checklist item is not a task as it doesn't include the global filter
```

And here is a query to show its files:

````
# Query: all tasks in ACME file

```tasks
path includes acme
```
````

And here is what it looks like for me, in all 3 views:

![image](https://user-images.githubusercontent.com/4840096/227878577-c9684fd3-29ba-4bf0-a7bc-af640ff390eb.png)

**Let's talk about this and the nearby tests when we chat later...**